### PR TITLE
chore: share dashboard auth state across apps

### PR DESCRIPTION
**Pull request by alaister**
## What kind of change does this PR introduce?

chore

## What is the current behaviour?

docs and www sites do not have access to the dashboard's auth state.

## What is the new behaviour?

the `common` package now contains the dashboard auth state, and can be accessed from any app the adds the provider and then imports the auth state hooks (`useAuth()` etc)

## Additional context

this PR also swaps the sign in / sign up buttons in the navbar with a dashboard button if the user is already signed in

Screenshot 2023-03-30 at 1 20 19 pm

### CODE REVIEW

File: apps/docs/pages/_app.tsx [hunk 1]
```diff
@@ -66,6 +66,20 @@ function MyApp({ Component, pageProps }: AppPropsWithLayout) {
       <Favicons />
       {IS_PLATFORM || LOCAL_SUPABASE ? (
```
Comment (joshenlim) [inline apps/docs/pages/_app.tsx:67]: not related to this PR but why do we have `LOCAL_SUPABASE` here? feels like its a dupe of `IS_PLATFORM`?
its only being used here in _app as well, unless LOCAL_SUPABASE !== IS_PLATFORM?

File: apps/www/components/Sections/APISection.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ function APISection(props: Props) {
     <div className="grid grid-cols-12 lg:gap-16">
       <div className="col-span-12 pb-8 lg:col-span-5 xl:col-span-5">
         <h2 className="h3">{props.title}</h2>
-        <p className="p">{props.text}</p>
+        <div className="p">{props.text}</div>
```
Comment (joshenlim) [inline apps/www/components/Sections/APISection.tsx:39]: just to sanity check this was intentional?

File: apps/docs/pages/_app.tsx (hunk 1 above)
Comment (joshenlim) [reply apps/docs/pages/_app.tsx:67]: there's no env var `NEXT_PUBLIC_LOCAL_SUPABASE` on vercel either - perhaps we can remove?

Comment (alaister) [reply apps/docs/pages/_app.tsx:67]: Hmm yeah I wasn't too sure about LOCAL_SUPABASE either. @gregnr are you the person to ask here?

File: apps/www/components/Sections/APISection.tsx (hunk 2 above)
Comment (alaister) [reply apps/www/components/Sections/APISection.tsx:39]: Yep. Sorry I should have left a comment on these seemingly random file changes. I was just fixes some hydration errors in www

File: apps/docs/pages/_app.tsx (hunk 1 above)
Comment (joshenlim) [reply apps/docs/pages/_app.tsx:67]: no worries - lets just leave it here for now and revisit it later

**Review by joshenlim — APPROVED** [review]

**Pull request merged**